### PR TITLE
Feature/refactor spatialos project settings merge

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR refactors some existing logic that retrieves the SpatialOS project from the spatial.json file and other utility functions from the FLocalDeploymentManager to the FSpatialGDKServicesModule and making them static. It also removes duplicated code from the Cloud settings that retrieves the SpatialOS project name.

#### Primary reviewers
@mattyoung-improbable @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp [hunk 1]
```diff
@@ -322,7 +242,7 @@ bool FLocalDeploymentManager::TryStopLocalDeployment()
 	bool bSuccess = false;
 
 	TSharedPtr<FJsonObject> SpotJsonResult;
-	bool bPasingSuccess = ParseJson(SpotDeleteResult, SpotJsonResult);
+	bool bPasingSuccess = FSpatialGDKServicesModule::ParseJson(SpotDeleteResult, SpotJsonResult);
```
Comment (Zomtorg) [inline SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: The `bPasingSuccess` typo has reemerged 💀

Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: :o how strange. I based the branch on the one which was merged with the RC. Hope nothing else has reemerged

Comment (Zomtorg) [reply SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: Down below the `bPasingSuccess` is actually correct - might be just me fixing only one of them

**Comment by girayozil** [comment]
Beautiful

File: SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp (hunk 1 above)
Comment (joshuahuburn) [reply SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp]: I've actually fixed this in a different branch so let's worry about it there :D

**Pull request merged**